### DISCUSSION TITLE
Slightly uncomfortable autocorrect

### DISCUSSION THREAD

**Discussion by snakysnake** [prisma/prisma#22814] [Q&A] [answered]
## Problem

I have a model called "PDF". 
When I try to access it using prisma.pdf.whatever() it's suggesting me to use prisma.pDF.whatever()

## Suggested solution

Dont suggest different spelling on fully uppercased model names

## Alternatives

Dont allow uppercase spelling to begin with, although it makes sense on models such as PDF

## Additional context

![slighty-disturbing-message](https://github.com/prisma/language-tools/assets/46492517/7047113a-547a-4d65-8194-700030df2024)

**Comment by janpio** [comment]
(I moved this out of the lanugage tools repository, as this is in no way influenced by our language tools or VS Code extension, but just the default autocompletion of our generated Client)

That is currently just how Prisma works, unfortunately. The JS API we generate lowercases all first characters of all models. There is no way to change that right now.

**Comment by snakysnake** [comment]
Okay. I have another question. It is recommended to only create a PrismaClient once in your application flow.
In my case I am adding the PrismaClient to my request object and passing that on. 
Here, the VSCode Extension does not recognize this object as a PrismaClient, thus I don't have any autocompletion.
Do you think it's sensible to create an issue for this somewhere, or is something like this already in the pipeline?
Dear regards, snakysnake

**Reply by janpio** [reply, marked as answer]
Autocompletion for Prisma Client is done by the TypeScript language server, not by the Prisma VS Code extension. That only adds autocompletion and similar feature for Prisma schemas.

But yes, if something with the Prisma Client types does not work as expected it is totally reasonable to open a discussion here so that our support colleagues can try to help you with that.
